migrations/authgear/20240327130255-add_reindexing_timestamps.sql

```diff
@@ -0,0 +1,9 @@
+-- +migrate Up
+ALTER TABLE _auth_user ADD COLUMN last_indexed_at timestamp without time zone;
+ALTER TABLE _auth_user ADD COLUMN require_reindex_after timestamp without time zone;
+UPDATE _auth_user SET require_reindex_after = updated_at;
+ALTER TABLE _auth_user ALTER COLUMN require_reindex_after SET NOT NULL;
```

Comment:
> Why do we want to make it NOT NULL? If require_reindex_after is NULL, then the reindex task can just skip.

1. I think it is simpler if we don't need to handle null case, or we need to remember the meaning of "null" (It means reindexing is required or not required?).
2. If we allow null value in the column and skip the reindexing on null, and in the future we made bugs that updated the user but forgot to update this timestamp. The user could become never indexed. Alternatively, if we treat null as "always update", at least the user will still be updated and effectively it will be same as before this pr. But "always update" can actually be done by a always in past timestamp such as 1970-01-01, so I concluded that null is not necessary. 

> One thing to note, when reindex is triggered by CLI, then it should always reindex regardless of these two columns.

I think that is the current behavior, but it will still update `last_indexed_at`.

Reply:
> My concern is that when this migration is run, it appears that all users need reindex but actually it is not.

But this is actually what I intent to do. Before this pr, all users are "need to be reindexed" because the reindex task won't skip any reindexing. I want the users to keep this state, until the new reindexing task runs at least one time.

This avoids problem like, before this pr was deployed, there are 1 remaining reindexing jobs in the queue. After this pr was deployed, the 1 job will be skipped because the user are marked to "up to date" automatically after the migration, and result in an out of sync user.

Reply:
ok. Let me update the code to handle null.